### PR TITLE
test(common/web): unit tests for kvk-file-writer

### PR DESCRIPTION
**Pull request by markcsinclair**
Unit tests for `kvk/kvk-file-write.ts`

See #9052 

@keymanapp-test-bot skip

### CODE REVIEW

**Comment by ermshiperete** [comment]
Oh no, another race! :smile:  If #12709 gets merged first you'll have to rename the new test files to match the new `*.tests.ts` naming pattern.

**Comment by markcsinclair** [comment]
> Oh no, another race! 😄 If #12709 gets merged first you'll have to rename the new test files to match the new `*.tests.ts` naming pattern (and `tests` location).

File name changed to `kvk-file-writer.tests.ts` during merge of master

**Comment by markcsinclair** [comment]
There are problems with null handling of `associatedKeyboard`, key `ansiFont.name` and `unicodeFont.name` that will require a fix, plus incorrect initialisation of `BUILDER_KVK_STRING`, which should be `{ len: 1, str: '' }` not `{ len: 0, str: '' }`.

File: common/web/types/tests/kvk/kvk-file-writer.tests.ts [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+  flags: BUILDER_KVK_KEY_FLAGS = BUILDER_KVK_KEY_FLAGS.kvkkBitmap,
+  shift: BUILDER_KVK_SHIFT_STATE = BUILDER_KVK_SHIFT_STATE.KVKS_NORMAL,
+  text:  string = "text",
+  bitmap: number[] = null,
+): VisualKeyboardKey {
+  const vkk: VisualKeyboardKey = {
+    vkey: vkey,
+    flags: flags,
+    shift: shift,
+    text: text,
```
Comment (mcdurdin) [inline common/web/types/tests/kvk/kvk-file-writer.tests.ts:275]: totally a nit

```suggestion
    vkey,
    flags,
    shift,
    text,
```

**Review by mcdurdin — APPROVED** [review]
LGTM

File: common/web/types/tests/kvk/kvk-file-writer.tests.ts [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+        "associatedKeyboard",
+        DEFAULT_KVK_FONT,
+        DEFAULT_KVK_FONT,
+        undefined,
+      );
+      const writer = new KvkFileWriter;
+      const binary: BUILDER_KVK_FILE = writer['build'](vk);
+      checkBuilderKvkFile(binary, vk);
+    });
+    // it('can handle a null associatedKeyboard', () => {
```
[the remark this answers is not in the document]
Comment (markcsinclair) [reply common/web/types/tests/kvk/kvk-file-writer.tests.ts:71]: The tests were ready for the fix, so straightforward to shift to `it.skip()` ... I like it, hadn't seen it before ... reports `3 pending`, which is better than commenting out.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.157-alpha](https://keyman.com/downloads/releases/alpha/18.0.157)

**Pull request merged**